ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/reloperators/HiveTableScan.java

```diff
@@ -202,6 +203,12 @@ public HiveTableScan copyIncludingTable(RelDataType newRowtype) {
   // Also include partition list key to trigger cost evaluation even if an
   // expression was already generated.
   @Override public RelWriter explainTerms(RelWriter pw) {
+    if (pw instanceof RelTreeSignatureWriter) {
```

Comment:
@okumin, why don't you define RelWriter var and extract the common initialization part to avoid code duplication.

Reply:
Thanks for the suggestion. I have no objections. Could you please check if this update follows your suggestion?
https://github.com/apache/hive/pull/5077/commits/bd22d94e3cf1bc4375e7f91c91026ecc036b0ac6